Adds Summon Cheese

Sealed101 opened the pull request:
oh apparently this is my 100th PR on tg, which is weird because github reports 99 total, and i made at least one to the old voidcrew repo, and filtering tg prs by my name still shows 99. weird. here's to 100 more i guess?

<sub>could have been better if it was a get, jhonflupwliiard watch ur back 🔫 </sub>

## About The Pull Request

Adds a new spell granter book to the Wizard's Den - Summon Cheese, which grants the spell by the same name, which summons 9 heads of cheese. That's about it, I think. 

## Why It's Good For The Game

Wizards are a hungry bunch, so why can't they just summon food? They can even share, if they'd like, since the notion of a friendly wizard still exists

<details>
<summary>... </summary>

alright fine
i'm slightly malding over not getting the 77777 get so no more roleplaying in the pr body

Wizard Grand Ritual now has a hidden goal of sacrificing 50 cheese wheels. Sacrificing a cheese wheel is done with invoking the grand rune, and each invocation/pulse of the rune will whisk away more cheese until all cheese on the rune is taken by whichever entity lurks in the other realm. The sacrifice will be hinted at in an _ancient parchment_ which will be on the bookshelf (when i add it lmao) alongside the spell book. 

Meeting this cheese goal will lock the wizard's grand finale rune and grand finale effect to special ones. The cheese rune is mostly identical to the normal grand rune, barring the custom sprites/animations. 
The cheese finale consists of the wizard getting permanent Levitation (nogravity + free space movement), a 0.5 modifier(reducing incoming damage in half) to every damage type, a comically strong mood buff and **The Wabbajack**(separate sprite pending) - a juiced up chaos staff which can fire a lot more projectiles than a normal one can (it will get more, I may even make a few just for it). 
Everybody else (including any invader antags) gets hallucinations, 175 brain damage and a comically strong mood debuff, as well as a vendetta against the wizard. If the victim was already suffering from hallucinations from a trauma (including the quirk), they are instead healed.

if you didn't catch the obvious reference, this entire shtick is a tribute to Sheogorath. the rune makes use of daedric script, and most of the catchphrases are a direct quotation of the Daedric Prince of Madness, so if any of those things can get us in trouble somehow, let me know. i will be sad but i will comply.

This shtick is intended as an easter egg, really, so I wasn't really planning on expanding the wizard grand finale into heretic paths thing or whatever.

Oh, and finale grand runes now allow the wizard to select the effect even if it's time-gated. If it is, you just won't be able to invoke it until the time comes. The rune will tell you how much time is left until you can invoke it. And grand finale runes with a finale effect selected also glow in the color of their effect. I also think I fixed some step of the grand rune animation not triggering but I'm not sure.

<details><summary>Some rune animations (some are subject to change)</summary>

![rune_cheese_activate](https://github.com/tgstation/tgstation/assets/75863639/62ae184d-b6fc-4883-a169-4d8ca7636b40)

![rune_cheese_flash_2](https://github.com/tgstation/tgstation/assets/75863639/619545c8-3c55-4264-bfa4-d40067ef7406)


</details> 

## Why It's Great For The Game

it's funny

</details> 

## Changelog


:cl: Sealed101, EBAT_BASUHA for spritework
add: Wizard's Den now has a book of Summon Cheese in the Studies Room
/:cl:

Sealed101 committed:
summon cheese spell
adds a granter book for it to the bookshelf in the wizard den
the spell, uh, summons cheese. that's it

Sealed101 committed:
What you will need to make 1 cheese of approx 700g
- 7 ½ Litres Low Fat Milk. (2% fat)
- 2 ml Calcium Chloride dissolved in ¼ cup cool water
- ½ teaspoon TA61 Thermophilic Culture
- ½ teaspoon LH100 Helvetic Culture
- ¼ teaspoon Lipase Powder dissolved in ¼ cup cool water
- 2 ml Rennet dissolved in ¼ cup cooled boiled water
- 300g Salt for brine
- Cheese Cloth or Butter Muslin
- 800g - 1kg Hard Cheese Basket
- Cheese Press

Sealed101 committed:
Method:
1. Clean and sterilise everything that will be used during the cheese making process!
2. Prepare the Starter. The day before cheese making, prepare a “Thermophilic Culture” starter using both the TA61 and LH100 cultures as described in “Cheese Making Basics”.
3. Preparing the Milk. Heat the milk to 32°C. Add the Calcium Chloride solution, then the prepared starter and the Lipase solution. Cover and allow to ripen for 30 minutes.

Sealed101 commented:
damn i was typing too slow boo-womp

fuck you i'm grabbing breakfast

Y0SH1M4S73R commented:
If only we ran our repo like goonstation does so we could put a hypothetical secret interaction behind the secrets repo that only select approved coders could access.

CCC23 commented:
https://youtu.be/sM8mH0RNHRY

Ticklezcat commented:
I'd say don't code when you're hungry, but this is beautiful, more people should code while hungry.

Sealed101 commented:
the breakfast note is because i just woke up and i was three minutes too late from getting pr number 77777 not because i was making this pr while hungry

Fikou commented on code/modules/antagonists/wizard/grand_ritual/grand_rune.dm:
```diff
@@ -154,13 +183,16 @@
 
 /obj/effect/grand_rune/proc/remove_rune()
-	new /obj/effect/decal/cleanable/grand_remains(get_turf(src))
+	var/remains_typepath = /obj/effect/decal/cleanable/grand_remains
+	if(istype(src, /obj/effect/grand_rune/finale/cheesy))
+		remains_typepath = /obj/effect/decal/cleanable/grand_remains/cheese
```
i'd rather the path spawned be a variable you override on the cheese rune

Fikou commented on code/modules/projectiles/guns/magic/staff.dm:
```diff
@@ -171,6 +171,34 @@
+
+/obj/item/gun/magic/staff/chaos/true_wabbajack/Initialize(mapload)
+	. = ..()
+	allowed_projectile_types += subtypesof(/obj/projectile/magic)
```
it already has magic projectiles, dont you want |=

HoodMilk commented:
Do you mind? I'm busy doing the fishstick. It's a very delicate state of mind!

Also the ritual healing people who already have RDS is hilarious and flavorful.

carlarctg commented:
90 minutes is way too much man that aint happening. besides, itll encourage degenerate behavior with the wizard spam recalling and dragging on a empty round just to get final cheese

how will this interact with the death of a lich?

Sealed101 commented:
> 90 minutes is way too much man that aint happening. besides, itll encourage degenerate behavior with the wizard spam recalling and dragging on a empty round just to get final cheese
> 
> how will this interact with the death of a lich?

this was capped at 90 minutes because initially i had a much more grandiose effect but i couldn't implement it without a buttload of animation work and lemon not placing me on the naughty list

what lich

Sealed101 committed:
4. Renneting. Gently stir in the diluted Rennet with an up and down motion for at least 2 mins. Cover and allow 30 mins to set or until you get a "clean break".
5. Testing for a Clean Break. Test for a “clean break” by sliding your knife into the curd at an angle and lifting some on the side of the blade. If the curd breaks cleanly around the knife and whey runs into the crack that is made, you have a “clean break”.
6. Cutting the Curd. Cut the curds (according to the method described in the “Cheese Making Basics” information sheet) into 6mm cubes, then heat at a rate of 2°C every 5 minutes until you reach 38°C, stirring often.
7. Cooking the Curds. Raise the temperature by 3°C every 5 minutes until the temperature reaches 50°C, stirring often. The curds should now be the size of a grain of rice, and they will squeak when chewed. Allow the curds to set for 5 minutes.

Sealed101 committed:
8. Drain the Whey. Carefully pour off the whey without losing any of the curd particles. Line the 1kg cheese basket with the cheese cloth (or butter muslin). Place the curds into the basket and press lightly with 2.5kg for 15 minutes.
9. Redress the Cheese. Remove the cheese from the basket, carefully peel away the cheese cloth, turn the cheese over, re-dress it and press with 4kg for 30 minutes.
10. Repeat the process but press with 7kg for 2 hours.
11. Remove the cheese from the basket, line it with fresh cloth. Replace the cheese and press with 10kg for 12 hours.

Sealed101 commented:
i rgb'd your wabbajack

Sealed101 committed:
12. Brine the Cheese. Make a brine solution using 300g of Salt to 800ml of water. Remove the cheese from the basket, peel away the cheese cloth and soak in the brine for 24 hours at room temperature.

ATH1909 commented:
> this was capped at 90 minutes because initially i had a much more grandiose effect but i couldn't implement it without a buttload of animation work and lemon not placing me on the naughty list

since you decided to not go for that grandiose effect, can we reduce the minimum start time for the cheese ritual?

Sealed101 commented:
it's 45 minutes now, which it's like in the middle

Jacquerel commented on code/modules/antagonists/wizard/grand_ritual/grand_ritual_finale.dm:
```diff
@@ -399,6 +400,57 @@
+ * Already insane people are instead cured of their madness, ignoring any other effects as the station around them loses its marbles.
+ */
+/datum/grand_finale/cheese
+	/**we don't set name, desc and others, thus we won't appear in the radial choice of a normal finale rune */
```
```suggestion
	// we don't set name, desc and others, thus we won't appear in the radial choice of a normal finale rune
```
Using the `/** */` format will probably mess up the doc comments for the following proc

Jacquerel commented on code/modules/antagonists/wizard/grand_ritual/grand_rune.dm:
```diff
@@ -114,13 +119,39 @@
+	cheese_to_haunt = cheese_list.Copy(1, min(round(length(cheese_list) * (1 + times_invoked) * 0.33), max(length(cheese_list), 3)) + 1)
+	for(var/obj/item/food/cheese/wheel/sacrifice as anything in cheese_to_haunt)
+		sacrifice.touched_by_madness = TRUE
+		sacrifice.AddComponent(/datum/component/haunted_item, \
+		haunt_color = spell_colour, \
+		haunt_duration = 9 SECONDS, \
+		aggro_radius = 0, \
+		spawn_message = span_revenwarning("[sacrifice] begins to float and twirl into the air as it becomes enveloped in otherworldy energies!"), \
+		despawn_message = span_revenwarning("[sacrifice] descends on the ground one last time..."), \
+		)
```
```suggestion
		sacrifice.AddComponent(\
			/datum/component/haunted_item, \
			haunt_color = spell_colour, \
			haunt_duration = 9 SECONDS, \
			aggro_radius = 0, \
			spawn_message = span_revenwarning("[sacrifice] begins to float and twirl into the air as it becomes enveloped in otherworldy energies!"), \
			despawn_message = span_revenwarning("[sacrifice] descends on the ground one last time..."), \
		)
```

Jacquerel commented on code/modules/library/random_books.dm:
```diff
@@ -94,10 +94,22 @@
+	var/static/cheese_granter_spawned = FALSE
+
+/obj/structure/bookcase/random/reference/wizard/after_random_load()
+	if(!cheese_granter_spawned)
```
```suggestion
	if(cheese_granter_spawned)
		return
```

Sealed101 commented on code/game/objects/items/food/cheese.dm:
```diff
@@ -44,6 +44,8 @@
 	w_class = WEIGHT_CLASS_NORMAL
 	rat_heal = 35
+	///Whether this cheese wheel is about to be whisked away to a hellish realm or not
+	var/touched_by_madness = FALSE
```
i made this var to safeguard the rune from trying to sacrifice a cheese wheel twice. i suppose we could just apply the haunted trait with a different source but i'm not sure

Jacquerel replied:
What I mean is that the `haunted` trait is applied when it is sacrificed, so you can just check for that trait instead of this boolean.

Sealed101 replied:
it is removed when the item gets unhaunted, which it does for a second before being whisked away

Sealed101 commented on code/modules/antagonists/wizard/grand_ritual/fluff.dm:
```diff
@@ -0,0 +1,25 @@
+	starting_author = "Pelagius the Mad"
+	starting_title = "Worship and Reverence of the Divine Insanity"
+	starting_content = {"<html>
+			<head>
+			<meta http-equiv='Content-Type' content='text/html; charset=UTF-8'>
+			</head>
+			<body>
+			<i>Most of the scroll's contents are unintelligible, plagued with mold, milk stains and a stench of spolied goat cheese so potent,</i><br>
+			<i>you can barely resist turning your head to retch. What's left of the writings is vague and abstract, as if the author</i><br>
+			<i>was in a mad dash to pass on their findings.</i><br><br>
+			<i>However, the runes they have managed to scribe onto the parchment are oddly untouched by time, and remain distinct.</i><br>
+			<i>You also discover a schema for a more widely-used Grand Ritual rune, however it is dotted with yellow circles, which in turn are</i><br>
+			<i>filled with black dots. Are these supposed to be... <b>cheese wheels?..</b></i><br><br>
+			<i>As you finish skimming through the wreck that is this scroll, you hear a faint snicker somewhere beyond your mind's eye...</i><br>
+			</body>
+			</html>"}
```
my html-fu is limited and extremely rusty

Sealed101 committed:
13. Aging. Remove the cheese from the brine and pat dry. Place on a clean cheese board and store at 10-12°C and 85% humidity for at least 10 months. Turn daily for the first two weeks and then weekly after that. Remove any mould with a cloth dampened in vinegar or salt water.
14. Oil the Rind. After the cheese has aged for 2 months, rub the surface with olive oil to keep the rind on the cheese from drying.

Sealed101 committed:
This parmesan cheese recipe brought to you by:
https://www.countrybrewer.com.au/pages/how-to-make-parmesan-cheese.html
thank you for your service and please don't somehow sue us over a recipe that's available to the public
since i'm out of the recipe steps, i'm just going to resume my normalcommit messages
removes touched_by_madness. it's a bit of flavor lost but i don't particularly care for it

Sealed101 committed:
what the flarp!

Sealed101 committed:
summon cheese spell icon that's not just cheese

san7890 commented on code/datums/status_effects/buffs.dm:
```diff
@@ -454,3 +454,50 @@
+
+/datum/status_effect/blessing_of_insanity/on_apply()
+	if(ishuman(owner))
+		var/mob/living/carbon/human/human_owner = owner
+		human_owner.physiology.brute_mod *= 0.5
+		human_owner.physiology.burn_mod *= 0.5
+		human_owner.physiology.tox_mod *= 0.5
+		human_owner.physiology.oxy_mod *= 0.5
+		human_owner.physiology.clone_mod *= 0.5
+		human_owner.physiology.stamina_mod *= 0.5
```
cache the `physiology` var to a typed var like `var/datum/physiology/editable = human_owner.physiology` for cleanliness and whatever speed that might yield

san7890 replied:
i don't know the actual type of the physiology datum but you know what i meant. typecast cache stuffs

san7890 commented on code/datums/status_effects/buffs.dm:
```diff
@@ -454,3 +454,50 @@
+/datum/status_effect/blessing_of_insanity/on_remove()
+	if(ishuman(owner))
+		var/mob/living/carbon/human/human_owner = owner
+		human_owner.physiology.brute_mod *= 2
+		human_owner.physiology.burn_mod *= 2
+		human_owner.physiology.tox_mod *= 2
+		human_owner.physiology.oxy_mod *= 2
+		human_owner.physiology.clone_mod *= 2
+		human_owner.physiology.stamina_mod *= 2
```
ditto cache here

san7890 commented on code/game/objects/items/granters/magic/summon_cheese.dm:
```diff
@@ -0,0 +1,25 @@
+	action_name = "summon cheese"
+	granted_action = /datum/action/cooldown/spell/conjure/cheese
+	remarks = list(
+		"I never knew so many types of cheese existed...",
```
alphabetize unless you don't want to

san7890 commented on code/modules/antagonists/wizard/grand_ritual/grand_ritual_finale.dm:
```diff
@@ -399,6 +400,57 @@
+	priority_announce("Danger: Extremely potent reality altering object has been summoned on station. Immediate evacuation advised. Brace for impact.", "Central Command Higher Dimensional Affairs", 'sound/effects/glassbr1.ogg')
+
+	for (var/mob/living/carbon/human/crewmate as anything in GLOB.human_list)
+		if (!crewmate.mind)
```
`isnull()` here

san7890 commented on code/modules/library/random_books.dm:
```diff
@@ -94,10 +94,23 @@
+	if(cheese_granter_spawned)
+		return
+	cheese_granter_spawned = TRUE
+	new /obj/item/book/granter/action/spell/summon_cheese (src)
```
```suggestion
	new /obj/item/book/granter/action/spell/summon_cheese(src)
```

san7890 commented on code/modules/projectiles/guns/magic/staff.dm:
```diff
@@ -171,6 +171,46 @@
+		/obj/projectile/energy/declone,
+		/obj/projectile/bullet/honker,
+		/obj/projectile/bullet/mime,
+	) //if you ever try to expand this list, avoid adding bullets/energy porjectiles, this ain't supposed to be a gun... unless it's funny
```
```suggestion
	) //if you ever try to expand this list, avoid adding bullets/energy projectiles, this ain't supposed to be a gun... unless it's funny
```

san7890 commented on code/modules/projectiles/guns/magic/staff.dm:
```diff
@@ -171,6 +171,46 @@
+
+/obj/item/gun/magic/staff/chaos/true_wabbajack/Initialize(mapload)
+	. = ..()
+	allowed_projectile_types |= subtypesof(/obj/projectile/magic)
+	allowed_projectile_types |= subtypesof(/obj/projectile/temp)
+	allowed_projectile_types += subtypesof(/obj/projectile/bullet/rocket)
```
why do we have both `|=` and `+=` here

Sealed101 replied:
because fikou and i'm bad at computer

though we can just use |= everywhere i suppose, as to not dupe the usual chaos staff's pool

Sealed101 committed:
review tidbits
also added an ion projectile to The Wabbajack

Sealed101 committed:
Update code/modules/library/random_books.dm

Co-authored-by: san7890 <the@san7890.com>

Wallemations reviewed (approved):
<details>
 <summary> Spoiler </summary>

![68747470733a2f2f696462322e616666656374656461726330372e636f2e756b2f696d616765732f32353836323339392f37373737382f31373234343339373933393938323238383939392e706e67](https://github.com/tgstation/tgstation/assets/66052067/3d339114-279e-4888-9a57-0288b5045db9)
Holy shit W.D. Gaster
</details>

Sealed101 commented:
oh no
